.github/workflows/network.yml

```diff
@@ -0,0 +1,70 @@
+name: Network CI
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'lib/vdsm/network/**'
+      - 'lib/vdsm/common/network/**'
+      - 'tests/network/**'
+      - '.github/workflows/network.yml'
+  pull_request:
+    paths:
+      - 'lib/vdsm/network/**'
+      - 'lib/vdsm/common/network/**'
+      - 'tests/network/**'
+      - '.github/workflows/network.yml'
+
+jobs:
+  unit:
+    env:
+      IMAGE_TAG: ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [ centos-8, centos-9 ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install podman
+      - name: Run unit tests
+        run: sudo -E ./tests/network/unit/run-tests.sh
+
+  integration:
+    env:
+      IMAGE_TAG: ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [ centos-8, centos-9 ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install podman
+      - name: Run integration tests
+        run: sudo -E ./tests/network/integration/run-tests.sh
+
+  functional:
+    env:
+      IMAGE_TAG: ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [ centos-8, centos-9 ]
+        switch_type: [ ovs, linux-bridge ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install podman
+      - name: Run functional tests
+        run: |
+          sudo -E ./tests/network/functional/run-tests.sh \
+          --switch-type=${{ matrix.switch_type }} \
```

Comment:
If you change ./tests/network/functional/run-tests.sh to run all switch types when running
without arguments, the only difference in the jobs is the name of the test subdirectory.

Reply:
The configuration slightly differs for the switch types. And also having it running in one job would mean twice the time for that single job.

Reply:
Much better now! could be nice if we had one job with matrix to run all 3 types of tests but this can be cone later.